src/main/java/main/Storage.java

```diff
@@ -0,0 +1,185 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Manages the loading and saving of tasks to and from a file.
+ */
+public class Storage {
+    private final File file;
+    private final DateTimeFormatter fileFormatter = DateTimeFormatter.ofPattern("MMM d yyyy hh:mm a");
+
+    /**
+     * Constructs a Storage object with the specified file path.
+     *
+     * @param filePath The path to the file where tasks are stored.
+     */
+    public Storage(String filePath) {
+        this.file = new File(filePath);
+    }
+
+    /**
+     * Loads tasks from the file.
+     *
+     * @return An ArrayList of tasks loaded from the file.
+     */
+    public ArrayList<Task> load() {
+        ArrayList<Task> list = new ArrayList<>();
+        try {
+            // Load file if it exists
+            if (file.exists()) {
+                Scanner scanner = new Scanner(file);
+                while (scanner.hasNext()) {
+                    String line = scanner.nextLine();
+                    String[] parts = line.split("\\|");
+                    char type = line.charAt(0);
+
+                    switch (type) {
+                        case 'T':
+                            Todo todo = new Todo(parts[2]);
+                            if (parts[1].equals("X")) {
+                                todo.markTaskAsDone();
+                            }
+                            list.add(todo);
+                            break;
+                        case 'D':
+                            Deadline deadline = new Deadline(parts[2], LocalDateTime.parse(parts[3], fileFormatter));
+                            if (parts[1].equals("X")) {
+                                deadline.markTaskAsDone();
+                            }
+                            list.add(deadline);
+                            break;
+                        case 'E':
+                            Event event = new Event(parts[2], LocalDateTime.parse(parts[3], fileFormatter),
+                                    LocalDateTime.parse(parts[4], fileFormatter));
+                            if (parts[1].equals("X")) {
+                                event.markTaskAsDone();
+                            }
+                            list.add(event);
+                            break;
+                        default:
+                            throw new CorruptedFileException("File is corrupted.");
+                    }
+                }
+                System.out.println("Existing data file found. Data has been loaded.");
+            } else {
+                if (file.getParentFile().exists()) {
+                    file.createNewFile();
+                } else {
+                    file.getParentFile().mkdir();
+                    file.createNewFile();
+                }
+                System.out.println("No existing data file found. New data file \"dash.txt\" has been created.");
```

Comment:
confirm is chatgpt do for u one, dont lie i dont believe u so smart

Reply:
bruh i legit ownself did this u think chatgpt so smart??